src/main/java/bookstore/controller/BookController.java

```diff
@@ -39,27 +41,31 @@ public List<BookDto> getAll(Pageable pageable) {
     }
 
     @ResponseStatus(HttpStatus.CREATED)
+    @PreAuthorize("hasRole('ADMIN')")
     @PostMapping
     @Operation(summary = "Create a new book", description = "Create a new book")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @ResponseStatus(HttpStatus.OK)
+    @PreAuthorize("hasAnyRole('USER', 'ADMIN')")
```

Comment:
shouldn't admin user also have a role `USER`?

or you design your app in a way that admin does not have same priviliges as user has?